Sources/ProjectSpec/Settings.swift

```diff
@@ -28,14 +28,32 @@ public struct Settings: Equatable, JSONObjectConvertible, CustomStringConvertibl
             groups = jsonDictionary.json(atKeyPath: "groups") ?? jsonDictionary.json(atKeyPath: "presets") ?? []
             let buildSettingsDictionary: JSONDictionary = jsonDictionary.json(atKeyPath: "base") ?? [:]
             buildSettings = buildSettingsDictionary
-            configSettings = jsonDictionary.json(atKeyPath: "configs") ?? [:]
+
+            self.configSettings = try Self.validateMappingStyleInConfig(jsonDictionary: jsonDictionary)
         } else {
             buildSettings = jsonDictionary
             configSettings = [:]
             groups = []
         }
     }
 
+    private static func validateMappingStyleInConfig(jsonDictionary: JSONDictionary) throws -> [String: Settings] {
```

Comment:
This method is named `validate` but it returns a subset of the given JSONDictionary, it's a bit confusing. 

How about `extractValidConfigs(from jsonDictionary: JSONDictionary)`?